huaweicloud/services/dli/resource_huaweicloud_dli_queue.go

```diff
@@ -297,6 +351,62 @@ func filterByQueueName(body interface{}, queueName string) (r *queues.Queue, err
 		body)
 }
 
+func getQueuesByName(cfg *config.Config, region, elasticResourcePoolName, queueName string) ([]elasticresourcepool.QueueScalingPolicy, error) {
+	client, err := cfg.DliV3Client(region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DLI V3 client: %s", err)
+	}
+
+	opts := elasticresourcepool.ListElasticResourcePoolQueuesOpts{
+		ElasticResourcePoolName: elasticResourcePoolName,
+		QueueName:               queueName,
+		// The API document states that the default value of limit is 100, but if offset is specified without limit, paging will not take effect.
+		Limit: 100,
+	}
+	queueList, err := elasticresourcepool.ListElasticResourcePoolQueues(client, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(queueList) == 0 {
+		return nil, fmt.Errorf("error getting list of the queues (%s) to elastic resource pool (%s): %s",
+			queueName, elasticResourcePoolName, golangsdk.ErrDefault404{})
+	}
+
+	// The `queue_name` parameter supports fuzzy search.
+	for _, queue := range queueList {
+		if queue.QueueName == queueName {
+			return queue.QueueScalingPolicies, nil
```

Comment:
return queue object if you want to name this function as the `getQueueByName()` (not getQueuesByName) or rename this function as `getQueueScalingPoliciesByName`

Reply:
fixed